.travis.yml

```diff
@@ -1,15 +1,27 @@
-# Don't use any predefined language env
-# (see https://github.com/travis-ci/travis-ci/issues/4895#issuecomment-150703192)
+# We will use docker to set up out environment, so don't use any particular
+# language in Travis itself
 language: generic
 
+# Enable docker
 sudo: required
+services:
+    - docker
 
-services: docker
+# Don't fetch submodules recursively, just a shallow fetch
+git:
+    submodules: false
+before_install: git submodule update --init
 
 env:
-    - DIST=trusty
-    - DIST=xenial
+    global:
+        # Make sure beaver is in the PATH
+        - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
+    matrix:
+        - DIST=trusty
+        - DIST=xenial
 
-install: ci/travis-install.sh
+install: beaver install
 
-script: ci/travis-script.sh
+script:
+    - beaver docker run make -r all
+    - beaver docker run make -r deb
```

Comment:
This looks a bit too verbose actually to be something that will be used to run almost every command. Maybe I should just make it `beaver run`...